lib/active_model/serializer/collection_serializer.rb

```diff
@@ -46,7 +46,13 @@ def json_key
         # 3. get from collection name, if a named collection
         key ||= object.respond_to?(:name) ? object.name && object.name.underscore : nil
         # 4. key may be nil for empty collection and no serializer option
-        key && key.pluralize
+        key &&= key.pluralize
+        # 5. fail if the key cannot be determined
+        unless key
+          fail ArgumentError, "Cannot infer root key from collection type. Please specify the root or each_serializer option, or render a JSON String"
+        end
+
+        key
```

Comment:
how about?

`key or fail ArgumentError,  "Cannot infer root key from collection type. Please specify the root or each_serializer option, or render a JSON String"`

Reply:
Also note rubocop failure here ```lib/active_model/serializer/collection_serializer.rb:52:31: C: Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. (https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)
          fail ArgumentError, "Cannot infer root key from collection type. Please specify the root or each_serializer option, or render a JSON String"
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^```